Fully open-source AIE flow

jgmelber opened the pull request:
Use open source aie_api 
No longer rely on Vitis for any components to compile for AIE

Add aie_api as a submodule
Install aie_api with build 
Use aie_api from install for Peano programming examples and programming guide

keryell commented on programming_examples/makefile-common:
```diff
@@ -3,6 +3,8 @@ AIETOOLS_DIR ?= $(shell realpath $(dir $(shell which xchesscc))/../)
 AIE_INCLUDE_DIR ?= ${AIETOOLS_DIR}/data/versal_prod/lib
 AIE2_INCLUDE_DIR ?= ${AIETOOLS_DIR}/data/aie_ml/lib
 
+AIEOPT_DIR ?= $(shell realpath $(dir $(shell which aie-opt))/../)
```
```suggestion
AIEOPT_DIR ?= $(shell realpath $(dir $(shell which aie-opt))/..)
```

jgmelber replied:
Copy the entire contents of the directory in cmake

keryell commented on runtime_lib/aie_api/aieapi.cmake:
```diff
@@ -0,0 +1,13 @@
+  message("Installing aie_api includes from ${SRCPATH} in ${BUILDPATH}")
+
+  # copy header files into build area
+  install(DIRECTORY ${SRCPATH}/ DESTINATION ${BUILDPATH}/aie_api)
```
I am curious about all these trailing `/` in the project.

jgmelber replied:
Copy the entire contents of the directory in cmake

keryell replied:
Interesting. You mean that without a `/` it would only create an empty directory?

jgmelber replied:
```
A trailing slash will leave the last component empty and install the contents of the input directory to the destination. 
```

https://cmake.org/cmake/help/book/mastering-cmake/chapter/Install.html

keryell replied:
Interesting. With CMake I learn all my life long!
Actually this is the equivalent of using `/.` in as with normal Unix copy-related commands.

keryell reviewed (commented):
Could you improve the PR description (which will end up as the eventual commit message for the eternity) about why we need this new API?

jgmelber commented:
> Could you improve the PR description (which will end up as the eventual commit message for the eternity) about why we need this new API?

Changed the description and added a couple lines. We use this API in the single core code. It was previously distributed with Vitis. Now that this is open sourced and we can use Peano. We no longer have to depend on Vitis.

stephenneuendorffer commented on .gitmodules:
```diff
@@ -11,3 +11,6 @@
 	url = https://github.com/Xilinx/bootgen.git
+[submodule "third_party/aie_api"]
+	path = third_party/aie_api
+	url = https://github.com/jgmelber/aie_api
```
This is the best URL?

jgmelber replied:
For now, I have a fork that comments out an include to adf.h. A follow on will remove any defines to __AIENGINE__, that will gate the include. We use the deprecated convention that flag defines something about the version of the AIE version and redundantly define the correct __AIE_ARCH__.
